src/sentry/incidents/handlers/condition/anomaly_detection_handler.py

```
@@ -0,0 +1,94 @@
+                "enum": [sensitivity.value for sensitivity in AnomalyDetectionSensitivity],
+            },
+            "seasonality": {
+                "type": "string",
+                "enum": [seasonality.value for seasonality in AnomalyDetectionSeasonality],
+            },
+            "threshold_type": {
+                "type": "integer",
+                "enum": [threshold_type.value for threshold_type in AnomalyDetectionThresholdType],
+            },
+            "confidence": {
+                "type": "string",
+                "enum": [AnomalyType.HIGH_CONFIDENCE.value, AnomalyType.LOW_CONFIDENCE.value],
+            },
+        },
+        "required": ["sensitivity", "seasonality", "threshold_type", "confidence"],
+        "additionalProperties": False,
+    }
+
+    @staticmethod
+    def evaluate_value(update: DataPacket, comparison: Any) -> DetectorPriorityLevel:
+        sensitivity = comparison["sensitivity"]
+        seasonality = comparison["seasonality"]
+        threshold_type = comparison["threshold_type"]
+        confidence = comparison["confidence"]
+
+        subscription: QuerySubscription = QuerySubscription.objects.get(id=update.source_id)
+        data_source = DataSource.objects.filter(
+            source_id=update.source_id,
+        ).first()
+        if data_source is None:
+            # this should not happen
+            raise DetectorError("DataSource does not exist.")
+
+        subscription_update = update.packet
+        source_id = {
+            "data_packet_source_id": update.source_id,
+            "data_packet_type": data_source.type,
+        }
+
+        anomaly_data = get_anomaly_data_from_seer(
+            sensitivity, seasonality, threshold_type, subscription, subscription_update, source_id
+        )
+        # covers both None and []
+        if not anomaly_data:
+            # something went wrong during evaluation
+            raise DetectorError("Error during Seer data evaluation process.")
```

should this return False instead of raising an exception, if you're going to build exception handling for it? what's the outcome of the exception handling?

---

We want this to raise an error instead of returning False, because returning False indicates that we should set the detector priority level to OK.

I think we actually want to change this condition handler to emit multiple detector priority levels according to the anomaly detection result 🤔